src/client/websocket/WebSocketManager.js

```diff
@@ -79,6 +85,21 @@ class WebSocketManager {
     this.client.emit(Events.DEBUG, `[connection] ${message}`);
   }
 
+  async _handleSessionLimit(shard) {
+    if (!this.sessionStartLimit && this.spawnQueue.length) {
+      this.spawn();
+    } else if (this.sessionStartLimit) {
+      this.sessionStartLimit = await this.client.api.gateway.bot.get().then(r => r.session_start_limit);
```

Comment:
As for this comment in specific, I thought about that, but what if I am sharding on multiple machines? Won't both caches end up wholly incorrect? Also, you have to have 5 seconds in between identifies, otherwise you will end up with an invalid session. However in the cases where we get invalid sessions and there's nothing we can do about it, we already reconnect with a new session in such cases. It was also recommended to not cache them as they are subject to change (iirc).